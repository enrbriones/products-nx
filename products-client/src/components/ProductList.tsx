import React from 'react';
import { Product } from '../services/productService';

interface ProductListProps {
  products: Product[];
  onDelete: (id: string) => void;
}

const ProductList: React.FC<ProductListProps> = ({ products, onDelete }) => {

  if(!products.length){
    return <div className="p-4 mb-4 mt-8 text-sm text-blue-800 rounded-lg bg-blue-50 dark:bg-gray-800 dark:text-blue-400 text-center" role="alert">
    <span className="font-medium">No hay productos en la lista!</span>
  </div>
  }

  return (
  <div className='relative overflow-x-auto shadow-md sm:rounded-lg container flex justify-center'>
      <table className='w-full text-sm text-left rtl:text-right text-gray-500 dark:text-gray-400'>
        <thead className='text-xs text-gray-700 uppercase bg-gray-50 dark:bg-gray-700 dark:text-gray-400'>
            <tr className=''>
                <th className='px-6 py-3'>Nombre</th>
                <th className='px-6 py-3'>Descripci√≥n</th>
                <th className='px-6 py-3'>Precio</th>
                <th className='px-6 py-3'></th>
            </tr>
        </thead>
        <tbody>
        {products.map((product) => (
          <tr key={product.id} className='bg-white border-b dark:bg-gray-800 dark:border-gray-700'>
            <td className='px-6 py-4 font-medium text-gray-900 whitespace-nowrap dark:text-white'>{product.name}</td>
            <td className='px-6 py-4'>{product.description}</td>
            <td className='px-6 py-4'>${product.price}</td>
            <td className='px-6 py-4 flex justify-end'>
            <button onClick={() => onDelete(product.id)} style={{ marginLeft: '10px' }} className='bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-3 rounded'>Eliminar</button>
            </td>
          </tr>
        ))}
        </tbody>
      </table>
    </div>

  );
};

export default ProductList;
