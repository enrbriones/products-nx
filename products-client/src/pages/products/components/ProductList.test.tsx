// src/components/ProductList.test.tsx
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
// import '@testing-library/jest-dom/extend-expect'; // Para mÃ©todos adicionales como `toBeInTheDocument`
import ProductList from './ProductList';

describe('ProductList Component', () => {
  test('', () => {
    const {container}=render(<ProductList products={[]} onDelete={()=>{console.log('hola')}} />);
    console.log(container)
    expect(container).toMatchSnapshot()
  });
});
