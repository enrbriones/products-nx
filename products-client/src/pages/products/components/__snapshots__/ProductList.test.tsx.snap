// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component  1`] = `
<div>
  <div
    class="p-4 mb-4 mt-8 text-sm text-blue-800 rounded-lg bg-blue-50 dark:bg-gray-800 dark:text-blue-400 text-center"
    role="alert"
  >
    <span
      aria-label="productsnotfound"
      class="font-medium"
    >
      No hay productos en la lista!
    </span>
  </div>
</div>
`;
